test/functional/feature_signet.py

```diff
@@ -0,0 +1,85 @@
+]
+
+class SignetBasicTest(BitcoinTestFramework):
+    def generate(self, node, count):
+        height = node.getblockcount()
+        for _ in range(count):
+            addr = node.getnewaddress()
+            node.generatetoaddress(1, addr)
+        self.wait_until(lambda: node.getblockcount() == height + count, timeout=10)
+
+    def set_test_params(self):
+        self.chain = "signet"
+        self.num_nodes = 6
+        self.setup_clean_chain = True
+        shared_args1 = ["-signet_challenge=51"] # OP_TRUE
+        shared_args2 = []                         # default challenge
+        # we use the exact same challenge except we do it as a 2-of-2, which means it should fail
+        shared_args3 = ["-signet_challenge=522103ad5e0edad18cb1f0fc0d28a3d4f1f3e445640337489abb10404f2d1e086be430210359ef5021964fe22d6f8e05b2463c9540ce96883fe3b278760f048f5189f2e6c452ae"]
+
+        self.extra_args = [
+            shared_args1, shared_args1,
+            shared_args2, shared_args2,
+            shared_args3, shared_args3,
+        ]
+
+    def skip_test_if_missing_module(self):
+        self.skip_if_no_wallet()
+
+    def run_test(self):
+        self.log.info("basic tests using OP_TRUE challenge")
+        node = self.nodes[0]
+
+        self.log.info('getmininginfo')
+        mining_info = node.getmininginfo()
+        assert_equal(mining_info['blocks'], 0)
+        assert_equal(mining_info['chain'], 'signet')
+        assert 'currentblocktx' not in mining_info
+        assert 'currentblockweight' not in mining_info
+        assert_equal(mining_info['networkhashps'], Decimal('0'))
+        assert_equal(mining_info['pooledtx'], 0)
+
+        self.generate(node, 1)
+        tmpl = node.getblocktemplate({'rules': ['segwit', 'signet']})
+        self.log.info("getblocktemplate: Test capability advertised")
+        assert 'proposal' in tmpl['capabilities']
+        assert 'coinbasetxn' not in tmpl
+
+        self.log.info("pregenerated signet blocks check")
+        node = self.nodes[2]
+
+        height = 0
+        for block in signet_blocks:
+            assert_equal(node.submitblock(block), None)
+            height = height + 1
+            assert_equal(node.getblockcount(), height)
+
+        self.log.info("pregenerated signet blocks check (incompatible solution)")
+        node = self.nodes[4]
```

Comment:
I'd prefer to simply write out the node index. This might even be shorter

Reply:
That is fine by me.